extension/src-language-server/stpa/modelChecking/model-checking.ts

```diff
@@ -0,0 +1,171 @@
+                ltl += ">" + valueRange.secondValue;
+            } else if (valueRange.secondValue === "MAX") {
+                // MAX value is used
+                ltl += "<" + valueRange.firstValue;
+            } else {
+                // two range values are given without use of MIN or MAX
+                ltl = "(" + uca.vars[index].$refText + "<" + valueRange.firstValue + " && " + uca.vars[index].$refText + ">" + valueRange.secondValue + ")";
+            }
+        }
+    } else {
+        // value is in the given range
+        if (valueRange.secondValue === undefined) {
+            // only one value is given
+            if (valueRange.firstValue === "false") {
+                // we dont want to write "== false"
+                ltl = "!" + ltl;
+            } else if (valueRange.firstValue !== "true") {
+                ltl += "==" + valueRange.firstValue;
+            }
+        } else {
+            if (valueRange.firstValue === "MIN") {
+                // MIN value is used 
+                ltl += "<=" + valueRange.secondValue;
+            } else if (valueRange.secondValue === "MAX") {
+                // MAX value is used
+                ltl += ">=" + valueRange.firstValue;
+            } else {
+                // two range values are given without use of MIN or MAX
+                ltl = "(" + uca.vars[index].$refText + "<=" + valueRange.secondValue + " && " + uca.vars[index].$refText + ">=" + valueRange.firstValue + ")";
+            }
+        }
+    }
+    return ltl;
+}
+
+/**
+ * Creates the LTL string for the given arguments.
+ * @param rule The rule that should be translated to a LTL.
+ * @param contextVariables The string for the context variable values.
+ * @param controlAction The controlaction for the rule.
+ * @returns the LTL for the given arguments.
+ */
+function createLTLString(rule: Rule, contextVariables: string, controlAction: string): { formula: string, text: string; } {
+    let formula = "";
+    let text = "";
+    switch (rule.type) {
+        case "not-provided":
+            formula = "G ((" + contextVariables + ") -> (controlAction==" + controlAction + "))";
+            text = controlAction + " provided in context " + contextVariables;
+            break;
+        case "provided":
+            formula = "G ((" + contextVariables + ") -> !(controlAction==" + controlAction + "))";
+            text = controlAction + " not provided in context " + contextVariables;
+            break;
+        case "too-early":
+            formula = "G (((controlAction==" + controlAction + ") -> (" + contextVariables + ")) && !((controlAction==" + controlAction + ")U(" + contextVariables + ")))";
+            text = controlAction + " not provided too early in context " + contextVariables;
+            break;
+        case "too-late":
+            formula = "G (((" + contextVariables + ") -> (controlAction==" + controlAction + ")) && !((" + contextVariables + ")U(controlAction==" + controlAction + ")))";
```

Comment:
encapsulate the string building processes into own methods/objects for better readability